app/lib/activitypub/activity/create.rb

```diff
@@ -260,7 +260,7 @@ def process_attachments
     as_array(@object['attachment']).each do |attachment|
       media_attachment_parser = ActivityPub::Parser::MediaAttachmentParser.new(attachment)
 
-      next if media_attachment_parser.remote_url.blank? || media_attachments.size >= 4
```

Comment:
These references have been updated to use constants instead of hard-coded -- https://github.com/mastodon/mastodon/pull/30433 -- need rebase here.